### ISSUE TITLE
Communities: Authenticate window crash and visual issues

### ISSUE THREAD

**Issue by Cuteivist** [status-im/status-desktop#11531]
# Bug Report

## Description

Authenticate button proceed button is out of bounds and app crashes when canceling

Additional question here is, when validation popup is canceled the request is send anyway and it's also showing that request is pending (where request to join button is).

## Steps to reproduce

* Go to new community
* Press "Request to join"
* Fail biometrics
* Biometrics failed and password popups have buttons out of bounds

Here clicking outside the popup sends the request anyway without validation. 
This send was crashing the app.

* Validate correctly using password

Here sending request would also crash

#### Expected behavior

* Button should not be out of the bounds
* App should not crash

#### Side notes

* When I saw first message loaded int one of the chats clicking outside the popup did crash app less frequent

#### Actual behavior

<img width="1383" alt="image" src="https://github.com/status-im/status-desktop/assets/11396062/4874a0e5-db99-4ac3-ad8b-4362f89b6655">

<img width="1427" alt="image" src="https://github.com/status-im/status-desktop/assets/11396062/fc31eb0d-e7ed-4b51-8a7e-ebceeaf1ff61">

### Additional Information

- Status desktop version: 0.13rc7
- Operating System: MacOS M1

## Logs

<details>
<summary>When successfully authenticating</summary>

```
ERR 2023-07-14 09:03:00.650+02:00 error checking all channels permissions:   topics="community-service" tid=1790526 file=service.nim:1462 errMsg="key not found: 0x03073514d4c14a7d10ae9fc9b0f05abc904d84166a6ac80add58bf6a3542a4e50a4f431bd0-28d1-4d70-9871-301050b71ee3"
ERR 2023-07-14 09:03:00.907+02:00 rpc response error                         topics="rpc" tid=1790546 file=core.nim:36 err="\nstatus-go error [methodName:wakuext_checkPermissionsToJoinCommunity, code:-32000, message:unsuccessful request: 403 Forbidden ]\n"
ERR 2023-07-14 09:03:00.907+02:00 error doing rpc request                    topics="rpc" tid=1790546 file=core.nim:40 methodName=wakuext_checkPermissionsToJoinCommunity exception="\nstatus-go error [methodName:wakuext_checkPermissionsToJoinCommunity, code:-32000, message:unsuccessful request: 403 Forbidden ]\n"
ERR 2023-07-14 09:03:00.907+02:00 error checking permissions to join:        topics="community-service" tid=1790526 file=service.nim:1409 errMsg=
INF 2023-07-14 09:03:11.936+02:00 history request completed                  topics="mailservers-service" tid=1790526 file=service.nim:183
INF 2023-07-14 09:03:11.984+02:00 history request started                    topics="mailservers-service" tid=1790526 file=service.nim:178 numBatches=4
INF 2023-07-14 09:03:30.903+02:00 KeychainManager stopped:                   topics="keychain-service" tid=1790526 file=service.nim:58 errCode=1 errType=keychain errDesc="Could not retrieve private key from keystore: The specified item could not be found in the keychain"
Warning: libpng warning: iCCP: known incorrect sRGB profile (:0, )
INF 2023-07-14 09:03:35.710+02:00 history request completed                  topics="mailservers-service" tid=1790526 file=service.nim:183
ERR 2023-07-14 09:03:40.900+02:00 rpc response error                         topics="rpc" tid=1790549 file=core.nim:36 err="\nstatus-go error [methodName:wakuext_requestToJoinCommunity, code:-32000, message:unsuccessful request: 403 Forbidden ]\n"
ERR 2023-07-14 09:03:40.901+02:00 error doing rpc request                    topics="rpc" tid=1790549 file=core.nim:40 methodName=wakuext_requestToJoinCommunity exception="\nstatus-go error [methodName:wakuext_requestToJoinCommunity, code:-32000, message:unsuccessful request: 403 Forbidden ]\n"
Traceback (most recent call last)
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/nim_status_client.nim(193) nim_status_client
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/nim_status_client.nim(190) mainProc
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nimqml/src/nimqml/private/qobject.nim(49) qobjectCallback
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nimqml/src/nimqml/private/qobject.nim(34) onSlotCalled
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(563) onSlotCalled
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app_service/service/community/service.nim(1481) onAsyncRequestToJoinCommunityDone
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
[1]    3124 segmentation fault  /Applications/Status.app/Contents/MacOS/nim_status_client
```

</details>

<details>
<summary>When clicking outside the popup</summary>

```
ERR 2023-07-14 08:56:47.273+02:00 error checking all channels permissions:   topics="community-service" tid=1777928 file=service.nim:1462 errMsg="key not found: 0x03073514d4c14a7d10ae9fc9b0f05abc904d84166a6ac80add58bf6a3542a4e50a4f431bd0-28d1-4d70-9871-301050b71ee3"
ERR 2023-07-14 08:56:47.566+02:00 rpc response error                         topics="rpc" tid=1777949 file=core.nim:36 err="\nstatus-go error [methodName:wakuext_checkPermissionsToJoinCommunity, code:-32000, message:unsuccessful request: 403 Forbidden ]\n"
ERR 2023-07-14 08:56:47.567+02:00 error doing rpc request                    topics="rpc" tid=1777949 file=core.nim:40 methodName=wakuext_checkPermissionsToJoinCommunity exception="\nstatus-go error [methodName:wakuext_checkPermissionsToJoinCommunity, code:-32000, message:unsuccessful request: 403 Forbidden ]\n"
ERR 2023-07-14 08:56:47.568+02:00 error checking permissions to join:        topics="community-service" tid=1777928 file=service.nim:1409 errMsg=
INF 2023-07-14 08:56:47.573+02:00 history request completed                  topics="mailservers-service" tid=1777928 file=service.nim:183
INF 2023-07-14 08:56:47.629+02:00 history request started                    topics="mailservers-service" tid=1777928 file=service.nim:178 numBatches=4
INF 2023-07-14 08:56:49.331+02:00 history request completed                  topics="mailservers-service" tid=1777928 file=service.nim:183
2023-07-14T08:57:00+0200 unset [github.com/anacrolix/torrent.(*Client).NewAnacrolixDhtServer.func2:406]: dht server on 0.0.0.0:51053 (node id d9e14cc52f3ca3c3950780524f8022e080c5045e) completed bootstrap ({NumAddrsTried:58 NumResponses:24})
INF 2023-07-14 08:58:04.893+02:00 KeychainManager stopped:                   topics="keychain-service" tid=1777928 file=service.nim:58 errCode=1 errType=keychain errDesc="Could not retrieve private key from keystore: The specified item could not be found in the keychain"
Warning: libpng warning: iCCP: known incorrect sRGB profile (:0, )
signal 16 received but handler not on signal stack
fatal error: non-Go code set up signal handler without SA_ONSTACK flag

runtime stack:
runtime.throw({0x108668baa?, 0x14000100900?})
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/panic.go:1047 +0x40 fp=0x1400005a780 sp=0x1400005a750 pc=0x108529fa0
runtime.sigNotOnStack(0x10)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/signal_unix.go:1020 +0x5c fp=0x1400005a7a0 sp=0x1400005a780 pc=0x10854080c
runtime.adjustSignalStack(0x10, 0x1400004c000, 0x1400005a848)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/signal_unix.go:581 +0x250 fp=0x1400005a810 sp=0x1400005a7a0 pc=0x10853fb10
runtime.sigtrampgo(0x10, 0x1400005a9b0, 0x1400005aa18)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/signal_unix.go:469 +0x174 fp=0x1400005a890 sp=0x1400005a810 pc=0x10853f764
runtime.sigtrampgo(0x10, 0x1400005a9b0, 0x1400005aa18)
	<autogenerated>:1 +0x1c fp=0x1400005a8c0 sp=0x1400005a890 pc=0x10855a29c
runtime.sigtramp()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/sys_darwin_arm64.s:227 +0x4c fp=0x1400005a980 sp=0x1400005a8c0 pc=0x10855984c

goroutine 34 [syscall, locked to thread]:
runtime.goexit()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x1400005bfe0 sp=0x1400005bfe0 pc=0x108558a94

goroutine 17 [running, locked to thread]:
	goroutine running on other thread; stack unavailable

goroutine 2 [force gc (idle)]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000048fa0 sp=0x14000048f80 pc=0x10852c8e4
runtime.goparkunlock(...)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:369
runtime.forcegchelper()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:302 +0xb4 fp=0x14000048fd0 sp=0x14000048fa0 pc=0x10852c774
runtime.goexit()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000048fd0 sp=0x14000048fd0 pc=0x108558a94
created by runtime.init.6
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:290 +0x24

goroutine 3 [GC sweep wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000049770 sp=0x14000049750 pc=0x10852c8e4
runtime.goparkunlock(...)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:369
runtime.bgsweep(0x0?)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mgcsweep.go:278 +0xa4 fp=0x140000497b0 sp=0x14000049770 pc=0x10851a244
runtime.gcenable.func1()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mgc.go:178 +0x28 fp=0x140000497d0 sp=0x140000497b0 pc=0x10850e668
runtime.goexit()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x140000497d0 sp=0x140000497d0 pc=0x108558a94
created by runtime.gcenable
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mgc.go:178 +0x70

goroutine 4 [GC scavenge wait]:
runtime.gopark(0x14000070000?, 0x1086b1828?, 0x1?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000049f50 sp=0x14000049f30 pc=0x10852c8e4
runtime.goparkunlock(...)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:369
runtime.(*scavengerState).park(0x1087feaa0)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mgcscavenge.go:389 +0x5c fp=0x14000049f80 sp=0x14000049f50 pc=0x1085181fc
runtime.bgscavenge(0x0?)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mgcscavenge.go:617 +0x44 fp=0x14000049fb0 sp=0x14000049f80 pc=0x108518794
runtime.gcenable.func2()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mgc.go:179 +0x28 fp=0x14000049fd0 sp=0x14000049fb0 pc=0x10850e608
runtime.goexit()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x14000049fd0 sp=0x14000049fd0 pc=0x108558a94
created by runtime.gcenable
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mgc.go:179 +0xb4

goroutine 18 [finalizer wait]:
runtime.gopark(0x0?, 0x0?, 0x0?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:363 +0xe4 fp=0x14000044580 sp=0x14000044560 pc=0x10852c8e4
runtime.goparkunlock(...)
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/proc.go:369
runtime.runfinq()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mfinal.go:180 +0x128 fp=0x140000447d0 sp=0x14000044580 pc=0x10850d888
runtime.goexit()
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/asm_arm64.s:1172 +0x4 fp=0x140000447d0 sp=0x140000447d0 pc=0x108558a94
created by runtime.createfing
	/opt/homebrew/Cellar/go@1.19/1.19.9/libexec/src/runtime/mfinal.go:157 +0x94
Traceback (most recent call last)
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/nim_status_client.nim(193) nim_status_client
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/nim_status_client.nim(190) mainProc
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nimqml/src/nimqml/private/qobject.nim(49) qobjectCallback
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nimqml/src/nimqml/private/qobject.nim(34) onSlotCalled
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(564) onSlotCalled
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app/modules/shared_modules/keycard_popup/view.nim(104) onCancelActionClicked
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app/modules/shared_modules/keycard_popup/io_interface.nim(109) onCancelActionClicked
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app/modules/shared_modules/keycard_popup/module.nim(329) onCancelActionClicked
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app/modules/shared_modules/keycard_popup/internal/state.nim(143) executeCancelCommand
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app/modules/shared_modules/keycard_popup/internal/biometrics_password_failed_state.nim(21) executeCancelCommand
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app/modules/shared_modules/keycard_popup/controller.nim(575) terminateCurrentFlow
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app/modules/shared_modules/keycard_popup/controller.nim(447) cancelCurrentFlow
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/src/app_service/service/keycard/service.nim(146) cancelCurrentFlow
/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nim-keycard-go/keycard_go.nim(21) keycardCancelFlow
SIGABRT: Abnormal termination.
[1]    99810 abort      /Applications/Status.app/Contents/MacOS/nim_status_client
```

</details>

**Comment by caybro** [comment]
The overflowing button has been fixed in master too

**Comment by saledjenic** [comment]
@Cuteivist few things... looking into the logs, I cannot say it's keycard lib issue even we know that in some cases keycard lib crashes from some still unknown reason for us. I would rather say it's up to inappropriately handled response for request to join community or something like that, maybe access to property of a nil obj, or maybe direct access to prop of an json object which doesn't exist or something similar. `keycardCancelFlow` is usually called when we closing authenticate popup, even you don't use a keycard (maybe that's something we can improve) but we just call it to clear keycard lib requests, that's why you see the last line `/Users/jenkins/workspace/status-desktop/systems/macos/aarch64/package/vendor/nim-keycard-go/keycard_go.nim(21) keycardCancelFlow`, but I am not sure it's a culprit for the crash.

**Comment by anastasiyaig** [comment]
@Cuteivist autentication thing should be fixed in 0.14 (master) https://github.com/status-im/status-desktop/issues/11335

**Comment by lukaszso** [comment]
@Cuteivist i tried to reproduce this in latest master and was not able to:
1. The visual issues are gone
2. The request is not sent if authentication is cancelled by clicking outside the window so the app is not crashing as a result of that
